spec/support/global_state_reset.rb

```diff
@@ -0,0 +1,22 @@
+module OpenProject
+  ##
+  # Provides resets to any global state that may have been changed through tests
+  # and the change of which should not affect other tests.
+  module GlobalStateReset
+    module_function
+
+    def reset_global_state!
+      reset_current_user!
+    end
+
+    def reset_current_user!
+      User.current = User.anonymous
+    end
+  end
+end
+
+RSpec.configure do |config|
+  config.after(:suite) do
+    OpenProject::GlobalStateReset.reset_global_state!
```

Comment:
We don't technically need anything after []f76621(https://github.com/opf/openproject/commit/f7662130a96c1f36333a41db68b2c0d98e49750d). Originally I just added the `config.after(:suite) { User.current = ...` to `copy_project_job_spec.rb` to get rid of the "lazyness warning".

But I do think @ulferts's refactoring is good and would allow me and everyone else not to bother with that anymore. So I would like to go ahead and just cherry-pick that commit then.